### PR TITLE
fixing the _dot operation with return on blas1 interface.

### PR DESCRIPTION
**Pull request by mehdi-goli**
Fixing the _dot operation when return the value. Replacing the enum with namespace and static int as enum value cannot be detected by duplicator.

### CODE REVIEW

File: include/executors/executor_sycl.hpp [hunk 1]
```diff
@@ -44,7 +44,11 @@ namespace blas {
 /*!using_shared_mem.
  * @brief Indicates whether if the kernel uses shared memory or not.
  */
-enum class using_shared_mem { enabled, disabled };
+//enum class using_shared_mem { enabled, disabled };
```
Comment (Ruyk) [inline include/executors/executor_sycl.hpp]: Remove the commented code please

File: include/executors/executor_sycl.hpp [hunk 2]
```diff
@@ -44,7 +44,11 @@ namespace blas {
...
+namespace using_shared_mem {
+  static const int enabled=0;
+  static const int disabled=1;
+};
```
Comment (Ruyk) [inline include/executors/executor_sycl.hpp]: Can you add a comment explaining why this is required?

File: tests/blas1_interface_test.cpp [hunk 3]
```diff
@@ -125,7 +125,7 @@ int main(int argc, char* argv[]) {
       // EXECUTION OF THE ROUTINES
       _axpy<SYCL>(ex, bX.get_count(), alpha, bvX, 1, bvY, 1);
       _asum<SYCL>(ex, bY.get_count(), bvY, 1, bvR);
-      _dot<SYCL>(ex, bY.get_count(), bvX, 1, bvY, 1, bvS);
+      vS[0]=_dot<SYCL>(ex, bY.get_count(), bvX, 1, bvY, 1);
```
Comment (Ruyk) [inline tests/blas1_interface_test.cpp]: formatting

Comment (mehdi-goli) [inline tests/blas1_interface_test.cpp]: added

File: include/executors/executor_sycl.hpp (hunk 1 above)
Comment (mehdi-goli) [inline include/executors/executor_sycl.hpp]: done I put it as parts of explanation

File: include/executors/executor_sycl.hpp (hunk 2 above)
Comment (mehdi-goli) [inline include/executors/executor_sycl.hpp]: added

**Comment by Ruyk** [comment]
Thanks!

**Pull request merged**